lara-typescript/src/section-authoring/components/section-item-plugin-list.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from "react";
+import { ICreatePageItem, SectionColumns } from "../api/api-types";
+import { UserInterfaceContext } from "../containers/user-interface-provider";
+import { useGetAvailablePlugins, useGetPageItemEmbeddableMetaData,
+         useGetPageItemPlugins, usePageAPI } from "../hooks/use-api-provider";
+import { SectionItemPlugin } from "./section-item-plugin";
+
+import "./section-item-plugin-list.scss";
+
+interface ISectionItemPluginListProps {
+  sectionColumn: SectionColumns;
+  sectionId: string;
+  sectionItemId: string;
+}
+
+export const SectionItemPluginList: React.FC<ISectionItemPluginListProps> = ({
+  sectionColumn,
+  sectionId,
+  sectionItemId
+  }: ISectionItemPluginListProps) => {
+  const availablePlugins = useGetAvailablePlugins();
+  const pageItemPlugins = useGetPageItemPlugins(sectionItemId);
+  const pageItemEmbeddableMetaData = useGetPageItemEmbeddableMetaData(sectionItemId);
+  const { userInterface: {editingItemId}, actions: {setEditingItemId}} = React.useContext(UserInterfaceContext);
+  const { userInterface: {wrappedItemId}, actions: {setWrappedItemId}} = React.useContext(UserInterfaceContext);
+  const api = usePageAPI();
+  const [selectedPluginType, setSelectedPluginType] = React.useState<string>("");
+
+  const handlePluginSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedPluginType(e.target.value);
+  };
+
+  const handleAdd = () => {
+    if (selectedPluginType === "") {
+      alert("Please select a plugin type.");
+      return;
+    }
+    const itemPosition = 0;
+    setWrappedItemId(sectionItemId);
+    const newItem: ICreatePageItem = {
+      column: sectionColumn,
+      embeddable: selectedPluginType,
+      position: itemPosition,
+      section_id: sectionId,
+      wrapped_embeddable_id: pageItemEmbeddableMetaData.data?.embeddableId,
+      wrapped_embeddable_type: pageItemEmbeddableMetaData.data?.embeddableType
+    };
+    api.addPageItem(newItem);
+  };
+
+  const renderPluginsMenu = () => {
+    const plugins = availablePlugins.isSuccess ? availablePlugins.data.plugins.map((p: any, i: number) => {
+      const alreadyAdded = pageItemPlugins?.find(pi => pi.name === p.name);
+      if (alreadyAdded) {
+        return null;
```

Comment:
This is going to return an array with null objects and components, which is fine, but you could also `.filter` your list, and then `.map` your plugin components, roughly like this:
```
vailablePlugins.data.plugins.filter(p => pageItemPlugins?.find(pi => pi.name === p.name))
.map((p: any, i: number) => ... etc

```